qiskit/test/mock/utils/configurable_backend_v2.py

```diff
@@ -0,0 +1,183 @@
+                loc=qubit_frequency or 4.8, scale=std, size=n_qubits
+            ).tolist()
+
+        if not isinstance(qubit_readout_error, list):
+            qubit_readout_error = np.random.normal(
+                loc=qubit_readout_error or 0.04, scale=std, size=n_qubits
+            ).tolist()
+
+        if dt is None:
+            dt = 0.2222222222222222e-9
+
+        if not qubit_coordinates is None:
+            self.qubit_coordinates = qubit_coordinates
+
+        self.backend_name = name
+        self.description = description
+        self.gate_configuration = gate_configuration
+        self.measurable_qubits = measurable_qubits
+        self.qubit_t1 = qubit_t1
+        self.qubit_t2 = qubit_t2
+        self.qubit_frequency = qubit_frequency
+        self.qubit_readout_error = qubit_readout_error
+        self.single_qubit_gates = single_qubit_gates
+        self.std = std
+        # self.n_qubits = n_qubits
+
+        super().__init__(
+            name=self.backend_name,
+            description=self.description,
+            online_date=datetime.datetime.utcnow(),
+            backend_version="0.0.0",
+        )
+
+        self._target = Target(dt=dt)
+
+        # Add gates to target
+        # TODO: dynamic instruction properties
+        for gate, qubit_tuple_list in self.gate_configuration.items():
+            temp_gate_props = {
+                qubit_tuple: InstructionProperties(duration=0.0, error=0)
+                for qubit_tuple in qubit_tuple_list
+            }
+            if gate in parameterized_gates.keys():
+                self._target.add_instruction(
+                    gate(Parameter(parameterized_gates[gate])), temp_gate_props
+                )
+            else:
+                self._target.add_instruction(gate(), temp_gate_props)
+
+        # Add reset to target
+        reset_props = {
+            (i,): InstructionProperties(duration=3676.4444444444443)
+            for i in range(n_qubits)
+        }
+        self._target.add_instruction(Reset(), reset_props)
+
+        # Add Measurement properties to target
+        meas_props = {
+            (i,): InstructionProperties(
+                duration=3.552e-06, error=self.qubit_readout_error[i]
+            )
+            for i in self.measurable_qubits
+        }
+        self._target.add_instruction(Measure(), meas_props)
+
+        # Save qubit properties as member variable
+        self._qubit_properties = {
+            i: QubitProperties(
+                t1=self.qubit_t1[i],
+                t2=self.qubit_t2[i],
+                frequency=self.qubit_frequency[i],
+            )
+            for i in range(n_qubits)
+        }
+
+    @property
+    def target(self):
+        return self._target
+
+    @property
+    def max_circuits(self):
+        return None
+
+    @classmethod
+    def _default_options(cls):
+        return Options(shots=1024)
+
+    def run(self, run_input, **options):
+        raise NotImplementedError
```

Comment:
I think here we probably should have a comment `TODO: Remove after #7391 is implemented and this is based on FakeBackendV2`. Since I think this class being able to used as a simulation target is useful.